packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1040,52 +1032,34 @@ function packageEntryFailure(id: string, details?: string) {
   )
 }
 
-const conditionalConditions = new Set(['production', 'development', 'module'])
-
-function resolveExports(
-  pkg: PackageData['data'],
-  key: string,
-  options: InternalResolveOptionsWithOverrideConditions,
+/**
+ * This generates conditions that aren't inferred by `resolveExports`
+ * from the `options` object.
+ */
+function getInlineConditions(
+  options: InternalResolveOptions,
   targetWeb: boolean
 ) {
-  const overrideConditions = options.overrideConditions
-    ? new Set(options.overrideConditions)
-    : undefined
+  const inlineConditions: string[] = []
 
-  const conditions = []
-  if (
-    (!overrideConditions || overrideConditions.has('production')) &&
-    options.isProduction
-  ) {
-    conditions.push('production')
-  }
-  if (
-    (!overrideConditions || overrideConditions.has('development')) &&
-    !options.isProduction
-  ) {
-    conditions.push('development')
-  }
-  if (
-    (!overrideConditions || overrideConditions.has('module')) &&
-    !options.isRequire
-  ) {
-    conditions.push('module')
+  const conditions: readonly string[] =
+    options.overrideConditions || options.conditions
+
+  if (targetWeb) {
+    if (!conditions.includes('node')) {
+      inlineConditions.push('browser')
+    }
+  } else if (!conditions.includes('browser')) {
+    inlineConditions.push('node')
```

Comment:
This mirrors the `browser` option of lukeed's `resolve.exports` module ([see here](https://github.com/lukeed/resolve.exports/blob/587b0ba1adcbef257a108b7c8edb12ce0cb1bfb2/src/index.js#L73)), but we *might* want to replace the `['node']` expression with `[]` to avoid the assumption that non-browser environment is using Node.js